src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,640 @@
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char ***sv = data;
+        char *start;
+        char *end;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        /* XDG does not allow duplicate definitions. */
+        if (*sv) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Key %s was defined multiple times, ignoring.", lvalue);
+                return 0;
+        }
+
+        *sv = strv_new(NULL);
+        if (!*sv)
+                return log_oom();
+
+        /* We cannot use strv_split because it does not handle escaping correctly. */
+        start = skip_leading_chars(rvalue, NULL);
+
+        for (end = start; *end; end++) {
+                if (*end == '\\') {
+                        /* Move forward, and ensure it is a valid escape. */
+                        end++;
+                        if (strchr("sntr\\;", *end) == NULL) {
+                                log_syntax(unit, LOG_ERR, filename, line, 0, "Undefined escape sequence \\%c.", *end);
+                                return 0;
+                        }
+                        continue;
+                }
+
+                if (*end == ';') {
+                        _cleanup_free_ char *copy = NULL;
+
+                        copy = strndup(start, end - start);
+                        if (copy == NULL)
+                                return log_oom();
+                        r = xdg_unescape_string(unit, filename, line, copy);
+                        if (r < 0)
+                                return r;
+                        r = strv_consume(sv, TAKE_PTR(copy));
+                        if (r < 0)
+                                return r;
+
+                        start = end + 1;
+                }
+        }
+
+        /* Any trailing entry should be ignored if it is empty. */
+        if (end > start) {
+                r = strv_extend(sv, start);
+                if (r < 0)
+                        return r;
+        }
+
+        return 0;
+}
+
+static int config_item_xdg_lookup(
+                const void *table,
+                const char *section,
+                const char *lvalue,
+                ConfigParserCallback *func,
+                int *ltype,
+                void **data,
+                void *userdata) {
+
+        XdgAutostartService *service = userdata;
+        _cleanup_free_ char *lvalue_stripped = NULL;
+
+        assert(table == NULL);
+        assert(lvalue);
+        assert(func);
+        assert(ltype);
+        assert(data);
+
+        if (!streq_ptr(section, "Desktop Entry"))
+                return 0;
+
+        /* Space before and after the = sign should be ignored. */
+        lvalue_stripped = delete_trailing_chars(strdup(lvalue), NULL);
```

Comment:
actually, the delete_trailing_chars() is unnecessary, the whitespace is already removed by the config parser

Reply:
Hah, outch. Why the heck did I think it wouldn't?